### ISSUE TITLE
Unselectable menu items (in addition to active/inactive ones)

### ISSUE THREAD

**Issue by ggarra13** [fltk/fltk#1059]
**Is your feature request related to a problem? Please describe.**
Currently, menu items can be active or inactive.  One of the issues with inactive menu items is that they appear grayed out and can be hard to read.

**I am using FLTK**
 - 1.4 master

**Describe the solution you'd like**
I propose a FL_MENU_UNSELECTABLE, which would work as FL_MENU_INACTIVE (not selectable or hilighted), but that it would, however, display normally without being ghosted out as inactive does.

**Comment by ManoloFLTK** [comment]
My vote is -1.
Users will not understand why such menu items are inactive.
Greying out inactive items is standard.

**Comment by ggarra13** [comment]
See this picture:

![Captura de pantalla 2024-09-01 105824](https://github.com/user-attachments/assets/6d998ea9-097c-4f0f-ab58-61c6b984bb3c)

I have grayed out items, which are unselectable and are used as headings.

In the picture, ICS is set out to ACEScg.   However, this entry should not be selectable as an item, as it is just information.   I would likely also color it differently.  If I set it as inactive, it becomes the same as the heading and it is quite unreadable.

**Comment by Albrecht-S** [comment]
-1 :-1: as well from me, I agree with Manolo.

@ggarra13 The fact that your inactive menu items are hardly readabled is cause by the facts that
- FLTK grays out inactive items and
- your background color is dark

Obviously we can't apply `fl_contrast()` to inactive items since this might make inactive items brighter than active menu items.

A better solution would be to let the programmer choose the color(s) of active and inactive menu items, just like they can set background colors. This is nothing we can do quickly though, i.e. not in 1.4.x. We should consider this for FLTK 1.5 - when we may perhaps add an entire new menu system.

**Comment by MatthiasWM** [comment]
I use headlines in pulldown menus as well, for example in the FLUID alignment pulldown, and I wouldn't mind if they were better readable. But as Albrecht said, it's a 1.5 thing. I suggest we leave this issue open and get to it when we start 1.5 master.

**Comment by Rangi42** [comment]
Non-grayed-out "headlines" in menus would correspond to `<optgroup>` [headings](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/optgroup) in an HTML `<select>` dropdown. Styling them by default as bold, dedented, or something like that would indicate their nature as unselectable.

**Comment by MatthiasWM** [comment]
I added a headline menu item style and integrated it into Fluid. MacOS native menus recognize the bold font attribute for menus.
